app/models/network/technology.rb

```diff
@@ -29,14 +29,15 @@ def self.behaviors
     end
 
     def self.build(installed, profile, options)
-      new(installed, profile)
+      new(installed, profile, options)
     end
 
-    attr_reader :installed, :profile
+    attr_reader :installed, :profile, :options
 
-    def initialize(installed, profile)
+    def initialize(installed, profile, options)
```

Comment:
Instead of storing the entire `options` hash on every Technology, it's likely better to extract the values you care about and ignore the rest.

At the *very* least, `Hash#fetch` should be used so that an exception gets raised if an option key is missing:

```ruby
def initialize(installed, profile, options)
  super
  @capping_fraction = options.fetch(:capping_solar_pv_slider_value)
end
```

However, since we're targeting Ruby 2.1 and newer, I think this would be a good case for using keyword arguments. Classes which need to accept an option can do it with a keyword argument instead:

```ruby
class ConservingProducer
  def initialize(installed, profile, capping_fraction:, **)
    super
    @capping_fraction = capping_fraction
  end

  # ...
end
```

In the example, I called the option `capping_fraction` rather than `capping_solar_pv_slider_value` because the latter is a front-end term. Be as generic as possible when naming thing in the back-end code, because it may be reused for other purposes. You might want to have a wind turbine whose production is capped, and "solar_pv_slider_value" doesn't make much sense in that context.